pkg/extsvc/github/repos.go

```diff
@@ -379,6 +382,74 @@ query Repositories($ids: [ID!]!) {
 	return repos, nil
 }
 
+// GetRepositoriesByNameWithOwnerFromAPI fetches the specified repositories
+// (namesWithOwners) from the GitHub GraphQL API and returns a slice of
+// repositories.
+// If a repository is not found, it will return an error.
+//
+// The maximum number of repositories to be fetched is 30. If more
+// namesWithOwners are given, the method returns an error. 30 is not a official
+// limit of the API, but based on the observation that the GitHub GraphQL does
+// not return results when more than 37 aliases are specified in a query. 30 is
+// the conservative step back from 37.
+//
+// This method does not cache.
+func (c *Client) GetRepositoriesByNameWithOwnerFromAPI(ctx context.Context, token string, namesWithOwners []string) ([]*Repository, error) {
+	if len(namesWithOwners) > 30 {
+		return nil, errors.New("cannot fetch more than 30 repositories via GraphQL API")
+	}
+
+	query, err := c.buildGetRepositoriesBatchQuery(ctx, namesWithOwners)
+	if err != nil {
+		return nil, err
+	}
+
+	var result map[string]*Repository
+	err = c.requestGraphQL(ctx, token, query, map[string]interface{}{}, &result)
+	if err != nil {
+		if gqlErrs, ok := err.(graphqlErrors); ok {
+			for _, err2 := range gqlErrs {
+				if err2.Type == graphqlErrTypeNotFound {
+					continue
+				}
+				return nil, err
+			}
+		} else {
+			return nil, err
+		}
+	}
+
+	repos := []*Repository{}
```

Comment:
Declaring empty slices, usually, benefits from avoiding the eager allocation this incurs by using the `var repos []*Repository` syntax. See: https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices

Reply:
In this case, however, we already know the upper bound of the repos we're working with, so this would be my proposal:

```suggestion
	repos := make([]*Repository, 0, len(result))
```